covigator/references/lineage_annotation.py

```diff
@@ -1,27 +1,562 @@
+            site=variant_string,
+            variant_id=variant_id,
+            variant_type=VariantType.SNV.name,
+            position=hgvs.position,
+            reference=hgvs.reference,
+            alternate=hgvs.alternate,
+            protein=hgvs.protein,
+            level=VariantLevel.PROTEOMIC.name if hgvs.protein is not None else VariantLevel.GENOMIC.name,
+            hgvs_p=hgvs.hgvs_p,
+            length=1,
+            annotation=hgvs.annotation
+        )
+        return mutation_info
+
+    def _parse_nucleotide_level_deletion(self, variant_string, hgvs) -> MutationInfo:
+        """
+        Create MutationInfo representation of pangolin nucleotide level deletion description
+        """
+        variant_id = self._generate_variant_id(hgvs.protein, hgvs.position, hgvs.reference, hgvs.alternate)
+        mutation_info = MutationInfo(
+            site=variant_string,
+            variant_id=variant_id,
+            variant_type=VariantType.DELETION.name,
+            position=hgvs.position,
+            reference=hgvs.reference,
+            alternate=hgvs.alternate,
+            length=len(hgvs.reference),
+            protein=hgvs.protein,
+            level=VariantLevel.PROTEOMIC.name if hgvs.protein is not None else VariantLevel.GENOMIC.name,
+            hgvs_p=hgvs.hgvs_p,
+            annotation=hgvs.annotation
+        )
+        return mutation_info
+
+    def _parse_mutation_sites(self, variant_string: str) -> List[Dict]:
+        """
+        Parse mutation site definition from pango constellation files. This code
+        is based on the parser found in the utility script of scorpio. Mutations
+        on nucleotide level are parsed and translated to protein level.
+        """
+        snp_pattern = re.compile('([ACTG]+)([0-9]+)([ACTG]+)')
+        insertion_pattern = re.compile(r'(\w+):(\d+)\+([a-zA-Z]+)')
+        aa_mutation = re.compile(r'([a-zA-Z-*]+)(\d+)([a-zA-Z-*]*)')
+        mutation_list = []
+        this_mut = variant_string.split(":")
+        # Insertions will be skipped as there are only two lineage defining insertions for now. We insert them manually
+        # with a SQL script
+        if "+" in variant_string:
+            logger.warning("Skipping nucleotide insertion {}...".format(variant_string))
+        # Deletions (nucleotide level) => translated to protein level
+        elif this_mut[0] == "del":
+            hgvs = self.get_hgvs_from_nuc_deletion({"genomic_position": int(this_mut[1]), "length": int(this_mut[2])})
+            mutation_list.append(self._parse_nucleotide_level_deletion(variant_string, hgvs))
+        # SNPs (nucleotide level, can be non-synonymous, missense and intergenic) => translated to protein level
+        elif this_mut[0] in ["snp", "nuc"]:
+            match = re.match(snp_pattern, this_mut[1])
+            if not match:
+                logger.warning("Could not parse the following site definition: {}".format(variant_string))
+                return [None]
```

Comment:
Avoid returning `[None]`, return instead `[]`. You have some code explicitly dedicated to remove these `None` values (see other comment)